Imath/ImathBox.h

```diff
@@ -35,88 +35,122 @@
 #ifndef INCLUDED_IMATHBOX_H
 #define INCLUDED_IMATHBOX_H
 
-//-------------------------------------------------------------------
-//
-//	class Imath::Box<class T>
-//	--------------------------------
-//
-//	This class imposes the following requirements on its
-//	parameter class:
-//
-//	1) The class T must implement these operators:
-//			+ - < > <= >= =
-//	   with the signature (T,T) and the expected
-//	   return values for a numeric type.
-//
-//	2) The class T must implement operator=
-//	   with the signature (T,float and/or double)
-//
-//	3) The class T must have a constructor which takes
-//	   a float (and/or double) for use in initializing the box.
-//
-//	4) The class T must have a function T::dimensions()
-//	   which returns the number of dimensions in the class
-//	   (since its assumed its a vector) -- preferably, this
-//	   returns a constant expression.
-//
-//-------------------------------------------------------------------
-
 #include "ImathNamespace.h"
 #include "ImathVec.h"
 
 IMATH_INTERNAL_NAMESPACE_HEADER_ENTER
 
+///
+/// The Box<T> template represents a bounding box defined by minimum and
+/// maximum values of type T. The min and max members are public.
+///
+/// The type T must implement:
+///
+///	1) Arithmetic operator + and - with the signature (T,T) and the expected return
+///        values for a numeric type.
+///
+///	2) Comparison operators <, >, <=, >= with the signature (T,T) and the
+///        expected return values for a numeric type.
```

Comment:
These comments just aren't correct (and weren't before, either).

A Box3f, say, is a `Box<Vec3<float>>`.

So the type T of a `Box<T>` must be a vector type. The T does *not* support `<`, `>`, `<=`, `>=`. But it *must* support `operator[]`, and whatever type `T[i]` returns must support the comparison operators.

Reply:
In the grand scheme of things, it would probably be better if the overall structure was like

```
template<class T, int Dim>
class Box {
public:
    using Point = Vector<T,Dim>;
    Point min, max;
    ...
};
```

and also set up a set of templated type aliases so that

```
Vector<T,2> = Vec2<T>
Vector<T,3> = Vec3<T>
```

etc.

There's a variety of template metaprogramming that this would make easier, and it would also clean up this issue here with Box where `Box<T>` is confusing in that the T signifies a vector template and not a basic data type like float.

As it is now, a user might be tempted to think that you could use the `Box` template to represent a 1D float interval.  But in fact, that wouldn't work, since there is no 1D vector type, and `Box<float>` would not expand to the right thing (it expects `T[i]` to work).

Reply:
Worth considering this change further, but I'll straighten out the comments in the meantime.